addon/-record-data-private/system/relationships/state/belongs-to.js

```diff
@@ -155,6 +157,17 @@ export default class BelongsToRelationship extends Relationship {
       return;
     }
     this.canonicalState = null;
+    /*
+      This isn't exactly correct because another record's payload
+      may tell us that this relationship is no longer correct
+      but that is not enough to tell us that this relationship is
+      now empty for sure. Likely we should be stale here but
+      that is probably a breaking change.
```

Comment:
Can you rephrase this?  what's an example payload that illustrates this?

Reply:
> However, while we are aware that <foo:1>.bar is no longer <bar:1>, we do not know with certainty that the canonical state of the relationship is now null and not some other bar.

What does this have to do with the second payload?  It is generally true that when the relationship of some other record changes this tells us nothing about any other relationship besides its inverse.

Reply:
Put another way, once we sever the connection between `<foo:1>` and `<bar:1>` any other information about `<bar:1>.foo` tells us nothing about `<foo:1>.bar` unless the value is `<foo:1>`

Of course something might have changed on the server and our client state could be wrong, but that's always the case.  The two flags being set represent client state; in the case you describe client state is still correct, no?  Which flag do you think is not correct in this scenario?